Build (pysam installation) is breaking?

Looks like all the tests that import pysam give this sort of error:

```
2023-05-03T05:25:54.0009511Z __________________ ERROR collecting strainflye/align_utils.py __________________
2023-05-03T05:25:54.0010265Z strainflye/align_utils.py:6: in <module>
2023-05-03T05:25:54.0010613Z     import pysam
2023-05-03T05:25:54.0011191Z /usr/share/miniconda/envs/strainflye/lib/python3.6/site-packages/pysam/__init__.py:5: in <module>
2023-05-03T05:25:54.0011636Z     from pysam.libchtslib import *
2023-05-03T05:25:54.0038126Z ERROR conda.cli.main_run:execute(47): `conda run make test` failed. (See above for error)
2023-05-03T05:25:54.0038623Z E   ImportError: libhts.so.2: cannot open shared object file: No such file or directory
```

And importing pysamstats also breaks things, but in a different way:

```
____________ ERROR collecting strainflye/tests/test_matrix_utils.py ____________
strainflye/tests/test_matrix_utils.py:8: in <module>
    import strainflye.matrix_utils as mu
strainflye/matrix_utils.py:8: in <module>
    from strainflye import (
strainflye/call_utils.py:6: in <module>
    import pysamstats
/usr/share/miniconda/envs/strainflye/lib/python3.6/site-packages/pysamstats/__init__.py:8: in <module>
    from .pileup import *
/usr/share/miniconda/envs/strainflye/lib/python3.6/site-packages/pysamstats/pileup.py:6: in <module>
    import pysamstats.opt as opt
E   AttributeError: module 'pysamstats' has no attribute 'opt'
```

Not sure what's up exactly -- the GitHub Actions logs are being weirdly vague about what versions of pysam/pysamstats were installed (they say ` ... (more hidden) ...` where i would expect to see info about these versions ._.) It looks like GitHub Actions is installing a relatively new version of bcftools (v1.17), and it looks like bcftools has conflicted with pysam before in a way that cause a similar libhts ImportError (https://github.com/pysam-developers/pysam/issues/427) -- maybe try pinning bcftools to < 1.16?

--- a/.github/workflows/main.yml
+++ b/.github/workflows/main.yml
@@ -43,14 +43,14 @@ jobs:
       - name: Install strainFlye (and pip dependencies)
         run: conda run -n strainflye pip install -e .[dev]
 
-      - name: Lint and stylecheck
-        run: conda run -n strainflye make stylecheck
-
       - name: Run tests
         run: conda run -n strainflye make test
 
       - name: Check that the strainFlye CLI seems good
         run: conda run -n strainflye strainFlye
+
+      - name: Lint and stylecheck
+        run: conda run -n strainflye make stylecheck
 
       - name: Upload code coverage information to Codecov
         uses: codecov/codecov-action@v2

--- a/environment.yml
+++ b/environment.yml
@@ -23,7 +23,7 @@ dependencies:
 
     - samtools >= 1.7
 
-    - bcftools
+    - bcftools < 1.17
 
     # Probably a later version will be installed; this is just the minimap2
     # version that added the asm20 preset. (If we want to use the more recently

--- a/environment.yml
+++ b/environment.yml
@@ -31,6 +31,8 @@ dependencies:
     - minimap2 >= 2.10
 
     # pysamstats requires pysam < 0.16, as of writing.
+    # the > 0.15 pin was added to fix the build -- otherwise, conda would
+    # install a really old pysam vsn (github.com/fedarko/strainFlye/issues/66)
     # also, we install pysam(stats) via conda, rather than pip, to simplify
     # installation (making it easier for pysam to "see" samtools) --
     # https://stackoverflow.com/a/60197930

--- a/environment.yml
+++ b/environment.yml
@@ -23,7 +23,7 @@ dependencies:
 
     - samtools >= 1.7
 
-    - bcftools < 1.17
+    - bcftools
 
     # Probably a later version will be installed; this is just the minimap2
     # version that added the asm20 preset. (If we want to use the more recently
@@ -34,7 +34,7 @@ dependencies:
     # also, we install pysam(stats) via conda, rather than pip, to simplify
     # installation (making it easier for pysam to "see" samtools) --
     # https://stackoverflow.com/a/60197930
-    - pysam < 0.16
+    - pysam > 0.15, < 0.16
 
     - pysamstats
 

--- a/setup.py
+++ b/setup.py
@@ -63,7 +63,10 @@ setup(
             "pytest >= 4.2",
             "pytest-cov >= 2.0",
             "flake8",
-            "black",
+            # black 22.10 stopped supporting python 3.6; ideally we'd adjust
+            # our CI to use later versions of black on the 3.7 build, but as a
+            # hacky solution pinning black is fine
+            "black < 22.10",
             # This isn't a hard version requirement -- I'm not sure what the
             # absolute minimum version is (maybe lower, probably not higher).
             # This should be good enough. (Needed for tutorial notebooks.)

